pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -449,7 +449,14 @@ func (c *Controller) performCapture(
 	if err != nil {
 		return false, err
 	}
-	pcapngWriter, err := pcapgo.NewNgWriter(file, layers.LinkTypeEthernet)
+
+	// set SnapLength here to make tcpdump on Mac OSX works. By default, its value is
+	// 0 and means unlimited, but tcpdump on Mac OSX will complain:
+	// 'tcpdump: pcap_loop: invalid packet capture length <len>, bigger than snaplen of 524288'
+	defaultNgInterface := pcapgo.DefaultNgInterface
+	defaultNgInterface.SnapLength = 524288
```

Comment:
doesn't this value relate to https://github.com/antrea-io/antrea/blob/b6d4238b1d4240f6040d396e2537f5edc4c2883a/pkg/agent/packetcapture/capture/pcap_linux.go#L32

should be use the same value (65536) given that this is what we used for the capture? Is there any special meaning to 524288?

Reply:
i'm not very if sets them as the same value is the best choice.

 For libbpf, 524288 is the default max value for all systems (ref to:  https://github.com/the-tcpdump-group/libpcap/blob/7812762d1d272026fb27bc0fcdfd156baaa0c13f/pcap-bpf.c#L1881). That's what the error message is shown. We can also set the `maxSnapshotBytes` to this value, but it allocate a buffer with this size for each capture. I mean for most user cases, we wouldn't use such a big value for it, it's pretty rase case for packet size > 65536, right? we can just return an error message for that in this case.

What do you think?

Reply:
My point is that if we set maxSnapshotBytes to 65536, then we will never capture more than 65536 bytes of packet data and there is no reason to set `SnapLength` to the max possible value in the pcap header. Tools reading the pcap may rely on this value to allocate a buffer (granted, 524288 bytes is not that large in the grand scheme of things). My understanding is that these 2 values mean the same thing essentially, so we could set them to the same value.

Reply:
sure